documentation/extensions/extensions.md

```diff
@@ -0,0 +1,31 @@
+# General Extensions to the GlobalPlatform TEE Internal API
+
+This document describes the OP-TEE extensions introduced with respect to the GlobalPlatform TEE API Specifications v1.0.
+
+Specific extensions documentation are part of:
+* Cryptographic Extensions
+  * [Concatenation Key Derivation](crypto_concat_kdf.md)
+  * [HMAC Key Derivation](crypto_hkdf.md)
+  * [Public-Key Key Derivation](crypto_pbkdf2.md)
+
+
+# Cache Maintenance Support
```

Comment:
Keep it in `documentation/extensions/extensions.md` just as you have done here. I too would like to avoid having too many documents. Separating things to some extent is good, but if we separate everything, then it will just be hard to find the information.